src/producer/async.js

```diff
@@ -0,0 +1,139 @@
+const grpc = require('grpc')
+const path = require('path')
+const uuid = require('uuid/v4')
+const MetricsReporter = require('./../lib/metricsReporter')
+const protoPath = path.resolve(__dirname, './protos/eventsgateway/grpc/protobuf/events.proto')
+const eventsProto = grpc.load(protoPath).eventsgateway
+
+class Async {
+  constructor(config, logger) {
+    this.config = config
+    this.address = this.config.grpc.serveraddress
+    if (!this.address) {
+      throw Error('no grpc server address informed')
+    }
+    this.grpcClient = new eventsProto.GRPCForwarder(
+      this.address, grpc.credentials.createInsecure()
+    )
+    this.logger = logger.child({ address: this.address })
+    this.metrics = new MetricsReporter(this.config)
+    this.method = '/eventsgateway.GRPCForwarder/SendEvents'
+    this.lingerIntervalMs = this.config.producer.lingerIntervalMs === undefined ?
```

Comment:
I'm not a fan of these long ternary lines. Maybe have a method like this (with better naming, please)

```
configure(field, default) {
  const val = this.config.producer[field] 
  this[field] = val === undefined ? default : val 
}
```

and inside the `constructor` use it:

 ```
this.configure('lingerIntervalMs', 500)
```

Reply:
great idea